test/conftest.py

```diff
@@ -47,14 +48,14 @@ def pytest_sessionstart(session) -> None:
 
 
 @pytest.fixture
-def input():
+def input() -> torch.Tensor:
```

Comment:
imo we should be using torch.tensor() rather than torch.Tensor(). Instantiating a tensor using `torch.Tensor` is the same as using `np.ndarray`. 

The `Tensor` class is generally used for type hinting, and that's it

Reply:
then it shouldn't take constructor args which build a Tensor 🤣 .  I'll take a look a look at this some more.  I spent some time in the docs especially looking at creating tensors with different d-types in the test code. 

This is in the draft follow on PR.

```python
@pytest.mark.parametrize("input", [
                          tensor([0, 0], dtype=int8),
                          tensor([0, 1.0, 2], dtype=int16),
                          tensor([2, 1, 0], dtype=int32),
                          tensor([ [0] ], dtype=int64),
                          tensor([ [0, 1.0] ], dtype=float16),
                          tensor([ [0, 1.0], [2, 3.0] ], dtype=float32),
                          tensor([ [3, 2], [1.0, 0] ], dtype=float64)
                        ])
def test_bmi_get_var_type(bmi_model_initialized, input):
    name = "precipitation"
    m = bmi_model_initialized
    m.input = input
    m._values[name] = m.input
    data = m.get_var_type(name)

    assert str(input.dtype).split(".")[1] == data
```

I'll see what it looks like to update the original uses of `Tensor()`